### PR TITLE
Conformance test for parameter-reference-like javascript

### PR DESCRIPTION
**Pull request by ThomasHickman**
This adds a conformance test to prevent errors like https://github.com/common-workflow-language/cwltool/issues/648.

### CODE REVIEW

File: v1.0/v1.0/parameter-reference.cwl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: "true"
+
+requirements:
+  - class: InlineJavascriptRequirement
+inputs:
+  - id: parameter
+    inputBinding:
+      valueFrom: $(true)
```
[the remark this answers is not in the document]
Comment (ThomasHickman) [reply v1.0/v1.0/parameter-reference.cwl]: Where does it say that in the specification? From a look through the CommandLineBindings (http://www.commonwl.org/v1.0/CommandLineTool.html#CommandLineBinding), it looks like `inputBinding.valueFrom` can return any type, at which the binding behaviour rules are applied

[the remark this answers is not in the document]
Comment (mr-c) [reply v1.0/v1.0/parameter-reference.cwl]: Sorry, you are correct. However this `inputBinding` is still invalid as it is for `type: boolean` has a `valueFrom` but lacks a `prefix`.

**Comment by mr-c** [comment]
Jenkins, okay to test

File: v1.0/v1.0/parameter-reference.cwl [hunk 2]
```diff
@@ -0,0 +1,14 @@
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: "true"
```
Comment (mr-c) [inline v1.0/v1.0/parameter-reference.cwl]: This test would be stronger if the correct behavior was communicated through the outputs, not merely through "not failing" :-) How about making the `baseCommand: echo` or use `args.py` like in https://github.com/common-workflow-language/common-workflow-language/blob/master/v1.0/v1.0/inline-js.cwl#L11 ?

**Comment by ThomasHickman** [comment]
^ I've moved the test to `inline-js.cwl`, as this is really testing a special case of Javascript expressions

**Pull request merged**